app/models/attachment_reference.rb

```diff
@@ -0,0 +1,61 @@
+class AttachmentReference < ActiveRecord::Base
+  after_initialize :set_expiry_at, if: proc { new_record? && !attachable }
+  before_validation :update_expiry_at
```

Comment:
@lowjoel I tried with the debugger, `attachable=` is never called if I create a material with FactoryGirl. I'll check again. 

It's always valid because there is a `before_validation`